articles/getting-started-with-dialogs-in-android-kotlin/index.md

```diff
@@ -0,0 +1,380 @@
+- Alert Dialog.
+- DatePicker Dialog.
+- TimePicker Dialog.
+- Dialog Fragment
+- BottomSheet Dialog.
+
+In this tutorial, we're going to lay our focus on `AlertDialog` where we'll mainly cover the following concepts:
+
+- Creating Alert dialogs;
+    - Adding title, buttons, message, and an icon
+    - Handle dialog events
+
+- Customizing Alert dialogs;
+    - Creating custom layout resource file
+    - Use of DataBinding to access views
+    - Creating a round-corner-shape dialog.
+
+#### Creating an Android project
+Launch Android Studio and create an `Empty Activity` project with the following specifications.
+
+`Name`: *Dialogs in Android*
+
+`Language`: *Kotlin*
+
+`Min-SDK`: *API-21*
+
+### Project setup
+#### Adding Dependencies and Plugins
+For us to use DataBinding, we have to enable and add the necessary plugin first. Open the app-level `build.gradle` file and add the following in the respective scope.
+
+```bash
+plugins {
+    id 'kotlin-kapt'
+}
+
+// inside the android block add the following
+
+buildFeatures{
+    dataBinding true
+}
+```
+*Sync* and wait for the Gradle build to finish. Once it's done, proceed to set up the UI.
+
+#### UI setup
+In our user interface, we need 2 buttons that we'll use to show the Dialogs. Open `activity_main.xml` file and paste the code below.
+
+```xml
+<Button
+    android:id="@+id/btnShowDefaultDialog"
+    android:layout_width="0dp"
+    android:layout_height="wrap_content"
+    android:layout_margin="10dp"
+    android:text="@string/show_dialog"
+    app:layout_constraintBottom_toBottomOf="parent"
+    app:layout_constraintEnd_toEndOf="parent"
+    app:layout_constraintStart_toStartOf="parent"
+    app:layout_constraintTop_toTopOf="parent"
+    app:layout_constraintVertical_bias="1" />
+
+<Button
+    android:id="@+id/btnShowCustomDialog"
+    android:layout_width="0dp"
+    android:layout_height="wrap_content"
+    android:layout_margin="10dp"
+    android:text="@string/show_custom_dialog"
+    app:layout_constraintBottom_toTopOf="@id/btnShowDefaultDialog"
+    app:layout_constraintEnd_toEndOf="parent"
+    app:layout_constraintStart_toStartOf="parent"
+    app:layout_constraintTop_toTopOf="parent"
+    app:layout_constraintVertical_bias="1" />
+```
+To fix the error about the unresolved reference in the text attribute, create strings in the strings resources as shown below.
+
+```xml
+<resources>
+    <string name="show_dialog">Show dialog</string>
+    <string name="show_custom_dialog">Show custom dialog</string>
+</resources>
+```
+*buttons preview*
+
+![buttons image](/engineering-education/getting-started-with-dialogs-in-android-kotlin/buttons.png)
+
+As mentioned earlier, we will use these buttons to show the necessary dialogs. To achieve this, we need DataBinding objects for each view in our layout. The beauty of DataBinding is that it generates the objects for us. We just need to enclose the layout with a `layout` tag as shown below.
```

Comment:
I don't think this is necessary. Constraint layout is also a layout. So, the binding class is also generated without adding the layout tag.